cmake/CompilerOptimizations.cmake

```diff
@@ -2,52 +2,54 @@
 check_c_compiler_flag(-ftree-vectorize HAS_TREE_VECTORIZE)
 check_cxx_compiler_flag(-ftree-vectorize HAS_CXX_TREE_VECTORIZE)
 if (HAS_TREE_VECTORISE)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ftree-vectorize")
+    add_compile_options("$<$<COMPILE_LANGUAGE:C>:-ftree-vectorize>")

```

Comment:
Is it really required to include these language checks? It's the same compiler for C/C++ anyways?

Reply:
On some platforms they might be a bit different I guess? For clang, there's `clang`  for c and `clang++` for c++.

Reply:
Moved to #1828